File: JumpGame_55.java
class Solution {
    public static boolean canJump(int[] nums) {
        int step = nums[0];
        int n = nums.length;
        for(int i = 1; i < n; i++){
            if(--step < nums[i] && step > 0) {
                step = nums[i];
            }
            if(step < 0) return false;
        }
        return true;
    }
    
    public static void main(String args[]) {
        int[] nums = {0,1};
        System.out.println(canJump(nums));
    }
}

File: JumpGame_55.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        length = len (nums)
        maxStep = nums[0]
        for i in range(1, length) :
            maxStep -= 1
            if maxStep < nums[i] and maxStep >= 0 :
                maxStep = nums[i]
            if maxStep < 0 :
                return False
        return True


if __name__ == '__main__':
    nums = [1,2,3]
    solu = Solution()
    print(solu.canJump(nums))
